docs/plugins/ast.md

```diff
@@ -0,0 +1,117 @@
+# Generate MyST AST in plugins
+
+A common usecase with plugins involves generating MyST AST and inserting it into the document.
+This page covers a few ways that you can do so.
+
+## Parse MyST markdwon to AST in a directive or role
+
+The easiest way to generate MyST AST in a plugin is by using the `parseMyst` function in the `ctx` variable. It may be easier to parse MyST Markdown into AST nodes rather than [using the MyST sandbox](sandbox.md) to preview them.
+
+Here's an example of using the `parseMyst` function within a directive plugin to parse the _argument_ of the directive into MyST AST:
+
+```{code} javascript
+:filename: src/myplugin.mjs
+const myDirective = {
+  name: "mydirective",
+  doc: "My new directive!",
+  arg: { type: String, doc: "To be parsed into MyST AST." },
+  run(data, vfile, ctx) {
+    const ast = ctx.parseMyst(data.arg);
+    return ast.children[0];
+  },
+};
+```
+
+`ctx.parseMyst` returns a `root` node, which contains the parsed MyST.
+This function takes the first child of the root node, which is usually better for inserting into the document.
+
+You can also use this to manually construct MyST outputs using the arguments of your directive as data. For example:
+
+```{code} javascript
+:filename: src/justacard.mjs
+const myDirective = {
+  name: "justacard",
+  doc: "Basically does what a card directive does!",
+  arg: { type: String, doc: "To be parsed into MyST AST." },
+  body: { type: String, doc: "The body of the directive." },
+  run(data, vfile, ctx) {
+    const ast = ctx.parseMyst(`:::{card} ${data.arg}\n${data.body}\n:::`);
+    return ast.children[0];
+  },
+};
+```
+
+:::{tip} If you need to use multi-line strings you must dedent them
+The example above puts a multi-line string onto one line by manually coding the `\n` characters.
+If you instead want to show a multi-line string in your code, for example like the following:
+
+
+```{code} javascript
+:filename: src/justacard.mjs
+const myDirective = {
+  name: "justacard",
+  doc: "Basically does what a card directive does!",
+  arg: { type: String, doc: "To be parsed into MyST AST." },
+  body: { type: String, doc: "The body of the directive." },
+  run(data, vfile, ctx) {
+    const ast = ctx.parseMyst(`
+    :::{card} ${data.arg}
+    ${data.body}
+    :::`);
+    return ast.children[0];
+  },
+};
+```
+
+You'll need to dedent the string manually to get the multi-line string to work.
+The easiest way to do this is to use the [`dedent` JavaScript package](https://www.npmjs.com/package/dedent).
```

Comment:
````suggestion
If you instead want to show a multi-line string in your code, you will need to remove the indentation manually, for example like the following:


```{code} javascript
:filename: src/justacard.mjs
const myDirective = {
  name: "justacard",
  doc: "Basically does what a card directive does!",
  arg: { type: String, doc: "To be parsed into MyST AST." },
  body: { type: String, doc: "The body of the directive." },
  run(data, vfile, ctx) {
    const ast = ctx.parseMyst(`
:::{card} ${data.arg}
${data.body}
::: `);
    return ast.children[0];
  },
};
```
````

I don't think we need to encourage using another package to work around a style preference -- zero indentation literals are fairly standard practice.